webrtc/rtcpeerconnection/rtcpeerconnection-constructor.html

```diff
@@ -0,0 +1,136 @@
+  '{ iceServers: [{ urls: [], credentialType: "token" }] }': false,
+  '{ iceServers: [{ urls: [], credentialType: "invalid" }] }': new TypeError,
+  // Blink and Gecko fall back to url, but it's not in the spec.
+  '{ iceServers: [{ url: "stun:stun1.example.net" }] }': new TypeError,
+
+  // iceTransportPolicy
+  '{ iceTransportPolicy: null }': new TypeError,
+  '{ iceTransportPolicy: undefined }': false,
+  '{ iceTransportPolicy: "relay" }': false,
+  '{ iceTransportPolicy: "all" }': false,
+  '{ iceTransportPolicy: "invalid" }': new TypeError,
+  // "none" in Blink and Gecko IDL, but not in the spec.
+  '{ iceTransportPolicy: "none" }': new TypeError,
+  // iceTransportPolicy is called iceTransports in Blink.
+  '{ iceTransports: "invalid" }': false,
+  '{ iceTransports: "none" }': false,
+
+  // bundlePolicy
+  '{ bundlePolicy: null }': new TypeError,
+  '{ bundlePolicy: undefined }': false,
+  '{ bundlePolicy: "balanced" }': false,
+  '{ bundlePolicy: "max-compat" }': false,
+  '{ bundlePolicy: "max-bundle" }': false,
+  '{ bundlePolicy: "invalid" }': new TypeError,
+
+  // rtcpMuxPolicy
+  '{ rtcpMuxPolicy: null }': new TypeError,
+  '{ rtcpMuxPolicy: undefined }': false,
+  '{ rtcpMuxPolicy: "negotiate" }': false,
+  '{ rtcpMuxPolicy: "require" }': false,
+  '{ rtcpMuxPolicy: "invalid" }': new TypeError,
+
+  // peerIdentity
+  '{ peerIdentity: toStringThrows }': new Error,
+
+  // certificates
+  '{ certificates: null }': new TypeError,
+  '{ certificates: undefined }': false,
+  '{ certificates: [] }': false,
+  '{ certificates: [null] }': new TypeError,
+  '{ certificates: [undefined] }': new TypeError,
+
+  // iceCandidatePoolSize
+  '{ iceCandidatePoolSize: toNumberThrows }': new TypeError,
+}
+for (const arg in argTests) {
+  const expr = 'new RTCPeerConnection(' + arg + ')';
+  const throws = argTests[arg];
+  if (throws) {
+    test(function() {
+      assert_throws(throws, function() {
+        eval(expr);
+      });
+    }, expr);
+  } else {
+    test(function() {
+      eval(expr);
+    }, expr);
+  }
+}
+
+promise_test(function() {
+  return RTCPeerConnection.generateCertificate({ name: "ECDSA", namedCurve: "P-256" })
+    .then(certificate => new RTCPeerConnection({ certificates: [certificate] }));
+}, 'new RTCPeerConnection({ certificates: [certificate] })');
+
+// initial state tests
+const initialState = {
+  'localDescription': null,
+  'currentLocalDescription': null,
+  'pendingLocalDescription': null,
+  'remoteDescription': null,
+  'currentRemoteDescription': null,
+  'pendingRemoteDescription': null,
+  'signalingState': 'stable',
+  'iceGatheringState': 'new',
+  'iceConnectionState': 'new',
+  'connectionState': 'new',
+  'canTrickleIceCandidates': null,
+  // TODO: defaultIceServers
+};
+for (const attr in initialState) {
+  test(function() {
+    const pc = new RTCPeerConnection;
+    assert_equals(pc[attr], initialState[attr]);
+  }, attr + ' initial value');
```

Comment:
This test seems strange. You're creating a set of peerconnections, but then you're only checking one attribute each round. Wouldn't it be easier / simpler / better to read something like

test(function()) {
    const pc = new RTCPeerConnection;
    for (const attr in initialState) {
        assert_equals(pc[attr], initialState[attr])
   }
 }, 'all initial values have their correct defaults');

The advantage I can see of the current state is that you get the attribute tested into the name of the test; the advantage of the other looping is that it actually tests that the attributes have the correct values at the same time - it's theoretically possible that a bad implementation could pass the first test but not the second, if signalingState was stable on run 5 and iceGatheringState was new on run 6, but not at the same time - would be completely perverse and unlikely, though.